hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OutboundResponseHandler.java

```diff
@@ -98,9 +99,14 @@ public void sendResponse(Operation operation, Object obj) {
         } else if (obj instanceof Throwable) {
             send = send(target, new ErrorResponse((Throwable) obj, operation.getCallId(), operation.isUrgent()));
         } else {
-            // most regular responses not wrapped in a NormalResponse. So we are now completely skipping the
-            // NormalResponse instance
-            send = sendNormalResponse(target, operation.getCallId(), 0, operation.isUrgent(), obj);
+            try {
+                // most regular responses not wrapped in a NormalResponse. So we are now completely skipping the
+                // NormalResponse instance
+                send = sendNormalResponse(target, operation.getCallId(), 0, operation.isUrgent(), obj);
```

Comment:
what about `else if (obj.getClass() == NormalResponse.class)` path?

Reply:
I wonder how can I write a test that will fail on that path ?
AtomicLAlterAndGetOperation seems like a good option. An operation that is backup aware and response type that I can make non-serializable. 

I will try to write a test for this and also fix the path you shared.

Reply:
@mdogan I have updated the pr to catch SerializationException from all send blocks, but I could not find a way to test it. I have tried DurableExecutor, AtomicReference in all cases serialization exception is thrown much earlier and this method is called with `ErrorResponse` as obj parameter.